src/azure-cli/azure/cli/command_modules/deploymentmanager/custom.py

```diff
@@ -268,46 +301,93 @@ def cli_service_unit_update(
     return instance
 
 
+def cli_service_units_list(
+        cmd,
+        resource_group_name,
+        service_topology_name,
+        service_name):
+
+    client = cf_service_units(cmd.cli_ctx)
+    return client.list(
+        resource_group_name=resource_group_name,
+        service_topology_name=service_topology_name,
+        service_name=service_name)
+
+
 def cli_step_create(
         cmd,
         resource_group_name,
-        step_name,
-        duration,
+        step_name=None,
+        step=None,
+        duration=None,
         location=None,
         tags=None):
 
-    waitStepProperties = WaitStepProperties(
-        attributes=WaitStepAttributes(duration=duration))
+    if (step is None and duration is None):
+        raise CLIError('usage error: specify either step or duration. If step is specified, it can either be a wait step or health check step.')  # pylint: disable=line-too-long
 
-    if location is None:
-        location = get_location_from_resource_group(cmd.cli_ctx, resource_group_name)
-
-    step = StepResource(
-        properties=waitStepProperties,
-        location=location,
-        tags=tags)
+    if (step is not None and duration is not None):
+        raise CLIError('usage error: specify only one of step or duration. If step is specified, it can either be a wait step or health check step.')  # pylint: disable=line-too-long
 
     client = cf_steps(cmd.cli_ctx)
+    if step is not None:
+        step_resource = get_step_from_json(client, step)
+        step_name = step_resource.name
+
+    elif duration is not None:
```

Comment:
In this scenario, if users don't provide step_name, which name will be used for your resource?

Reply:
The user will see a message saying that step_name is not provided. The validation will fail.